turb/python/stencils/turb_stencils.py

```diff
@@ -0,0 +1,3202 @@
+                tem = vcko[0, 0, 0] + vcko[0, 0, 1] - (v1[0, 0, 0] + v1[0, 0, 1])
+                f2[0, 0, 0][0] = f2[0, 0, 0][0] - tem * ptem1
+                f2_p1 = v1[0, 0, 1] + tem * ptem2
+            else:
+                f1_p1 = u1[0, 0, 1]
+                f2_p1 = v1[0, 0, 1]
+
+            if (
+                scuflg[0, 0]
+                and mask[0, 0, 0] >= mrad[0, 0]
+                and mask[0, 0, 0] < krad[0, 0]
+            ):
+                ptem = 0.5 * dsig * rdz * xmfd[0, 0, 0]
+                ptem1 = dtodsd * ptem
+                ptem2 = dtodsu * ptem
+                tem = ucdo[0, 0, 0] + ucdo[0, 0, 1] - (u1[0, 0, 0] + u1[0, 0, 1])
+                f1 = f1[0, 0, 0] + tem * ptem1
+                f1_p1 = f1_p1[0, 0] - tem * ptem2
+                tem = vcdo[0, 0, 0] + vcdo[0, 0, 1] - (v1[0, 0, 0] + v1[0, 0, 1])
+                f2[0, 0, 0][0] = f2[0, 0, 0][0] + tem * ptem1
+                f2_p1 = f2_p1[0, 0] - tem * ptem2
+
+        with interval(-1, None):
+            f1 = f1_p1[0, 0]
+            f2[0, 0, 0][0] = f2_p1[0, 0]
+            ad = ad_p1[0, 0]
+
+
+# Possible stencil name : moment_recover
+@gtscript.stencil(backend=backend)
+def moment_recover(
+    del_: FIELD_FLT,
+    du: FIELD_FLT,
+    dusfc: FIELD_FLT_IJ,
+    dv: FIELD_FLT,
+    dvsfc: FIELD_FLT_IJ,
+    f1: FIELD_FLT,
+    f2: FIELD_FLT_7,
+    hpbl: FIELD_FLT_IJ,
+    hpblx: FIELD_FLT_IJ,
+    kpbl: FIELD_INT_IJ,
+    kpblx: FIELD_INT_IJ,
+    mask: FIELD_INT,
+    u1: FIELD_FLT,
+    v1: FIELD_FLT,
+    conw: float,
+    rdt: float,
+):
+
+    with computation(FORWARD), interval(...):
+        utend = (f1[0, 0, 0] - u1[0, 0, 0]) * rdt
+        vtend = (f2[0, 0, 0][0] - v1[0, 0, 0]) * rdt
+        du = du[0, 0, 0] + utend
+        dv = dv[0, 0, 0] + vtend
+        dusfc = dusfc[0, 0] + conw * del_[0, 0, 0] * utend
+        dvsfc = dvsfc[0, 0] + conw * del_[0, 0, 0] * vtend
+
+    with computation(FORWARD), interval(0, 1):
+        hpbl = hpblx[0, 0]
+        kpbl = kpblx[0, 0]
+
+
+def mfpblt(
+    im,
+    ix,
+    km,
+    kmpbl,
+    ntcw,
+    ntrac1,
+    delt,
+    cnvflg,
+    zl,
+    zm,
+    q1_gt,
+    t1,
+    u1,
+    v1,
+    plyr,
+    pix,
+    thlx,
+    thvx,
+    gdx,
+    hpbl,
+    kpbl,
+    vpert,
+    buo,
+    xmf,
+    tcko,
+    qcko,
+    ucko,
+    vcko,
+    xlamue,
+    g,
+    gocp,
+    elocp,
+    el2orc,
+    mask,
+    qtx,
+    wu2,
+    qtu,
+    xlamuem,
+    thlu,
+    kpblx,
+    kpbly,
+    rbup,
+    rbdn,
+    flg,
+    hpblx,
+    xlamavg,
+    sumx,
+    scaldfunc,
+):
+
+    ce0 = 0.4
+    cm = 1.0
+    qmin = 1e-8
+    qlmin = 1e-12
+    alp = 1.0
+    pgcon = 0.55
+    a1 = 0.13
+    b1 = 0.5
+    f1 = 0.15
+    fv = 4.6150e2 / 2.8705e2 - 1.0
+    eps = 2.8705e2 / 4.6150e2
+    epsm1 = eps - 1.0
+
+    totflag = True
+
+    for i in range(im):
+        totflag = totflag and ~cnvflg[i, 0]
+
+    if totflag:
+        return kpbl, hpbl, buo, xmf, tcko, qcko, ucko, vcko, xlamue
```

Comment:
Yeah `cnvflg` isn't known until call time.  My thought about performance is that as is, this current way might be faster than having to evaluate `totflag` within stencils because of the overhead of calling multiple stencils and evaluate the if statement multiple times.  But like you said, we can do a profile to see which is faster.